File: math/equilibrium/multiplier.py
# Copyright (c) 2026 jmenichole. All rights reserved.
from equilibrium.constants import HOUSE_EDGE_DEN, HOUSE_EDGE_NUM


def next_multiplier_bps(current_bps: int, remain: int, fit: int) -> int:
    if fit <= 0 or remain <= 0:
        raise ValueError("ERR_ZERO_SURVIVE")
    return (current_bps * HOUSE_EDGE_NUM * remain) // (HOUSE_EDGE_DEN * fit)


def bps_to_x100(bps: int) -> int:
    return bps // 100

File: src/math/multiplier.ts
/**
 * Copyright (c) 2026 jmenichole. All rights reserved.
 */
import { HOUSE_EDGE_DEN, HOUSE_EDGE_NUM, MULTIPLIER_BPS_SCALE } from '../constants';

export function nextMultiplierBps(
  currentBps: number,
  remain: number,
  fit: number,
): number {
  if (fit <= 0 || remain <= 0) {
    throw new Error('ERR_ZERO_SURVIVE');
  }
  return Math.floor(
    (currentBps * HOUSE_EDGE_NUM * remain) / (HOUSE_EDGE_DEN * fit),
  );
}

export function payoutAmount(betAmount: number, multiplierBps: number): number {
  return Math.floor((betAmount * multiplierBps) / MULTIPLIER_BPS_SCALE);
}

export function displayMultiplier(multiplierBps: number): string {
  const hundredths = Math.floor(multiplierBps / 100);
  const whole = Math.floor(hundredths / 100);
  const frac = hundredths % 100;
  return `${whole}.${frac.toString().padStart(2, '0')}`;
}
